plugin/src/main/java/org/opensearch/ml/model/MLModelAndNodeManager.java

```diff
@@ -0,0 +1,294 @@
+            return;
+        }
+
+        // According to the node id to query the number of retries, if more than 2 (the maximum number of retries), we do not need to retry,
+        // that the number of unsuccessful reload has reached the maximum number of times, do not need to reload
+        Integer reTryTimes = 0;
+        if (isExistedIndex(ML_MODEL_RELOAD_INDEX)) {
+            reTryTimes = getReTryTimes(localNodeId);
+        }
+        if (reTryTimes > ML_MODEL_RELOAD_MAX_RETRY_TIMES) {
+            throw new RuntimeException("have exceeded max retry times, always failure");
+        }
+
+        // auto reload all models of this local node, if it fails, reTryTimes+1, if it succeeds, reTryTimes is cleared to 0
+        autoReLoadModelByNodeId(localNodeId);
+    }
+
+    @VisibleForTesting
+    void autoReLoadModelByNodeId(String nodeId) {
+        SearchRequest searchRequest = new SearchRequest(ML_TASK_INDEX);
+        SearchResponse response = client.execute(SearchAction.INSTANCE, searchRequest).actionGet();
+
+        if (response == null) {
+            return;
+        }
+        SearchHits searchHits = response.getHits();
+        if (searchHits == null) {
+            return;
+        }
+        SearchHit[] hits = searchHits.getHits();
+        if (CollectionUtils.isEmpty(hits)) {
+            return;
+        }
+
+        Integer reTryTimes = 0;
+        for (SearchHit hit : hits) {
+            try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, hit.getSourceRef())) {
+                ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                MLTask mlTask = MLTask.parse(parser);
+
+                String workerNode = mlTask.getWorkerNode();
+                MLTaskType mlTaskType = mlTask.getTaskType();
+                MLTaskState mlTaskState = mlTask.getState();
+
+                if (workerNode.contains(nodeId) && mlTaskType == MLTaskType.LOAD_MODEL && mlTaskState == MLTaskState.COMPLETED) {
+                    autoReLoadModelByNodeAndModelId(nodeId, mlTask.getModelId());
+                }
+            } catch (Exception e) {
+                reTryTimes++;
+                // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+                saveLatestReTryTimes(nodeId, reTryTimes);
+                log.error("ML Task id {} failed to parse, the reason is: {}", hit.getId(), e);
+                throw new RuntimeException(e);
+            }
+        }
+
+        // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+        saveLatestReTryTimes(nodeId, reTryTimes);
+    }
+
+    /**
+     *
+     * @param nodeId
+     * @param modelId
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeAndModelId(String nodeId, String modelId) {
+        MLLoadModelRequest mlLoadModelRequest = new MLLoadModelRequest(modelId, new String[] { nodeId }, false, false);
```

Comment:
We should add nodeIds here, since this function `client.execute(MLLoadModelAction.INSTANCE, ...)` will collect all ml nodes by default, this can cause unnecessary network calls.

Reply:
u can see ` new String[] { nodeId }` the string array is just has 1 element,it is our node that want to do reload operation,so it will not cause the effort of  model reloading by other node ids